covalent_dispatcher/_cli/service.py

```diff
@@ -320,21 +315,18 @@ def start(
         click.echo(MIGRATION_COMMAND_MSG)
         return ctx.exit(1)
 
+    update_config({"user_interface": {"port": port}, "dispatcher": {"port": port}})
+    if not no_cluster:
+        update_config(
+            {
+                "dask": {
+                    "mem_per_worker": mem_per_worker or "auto",
```

Comment:
I think `update_config` needs to be called for each of these options individually, if they are specified.  Otherwise specifying one of them will overwrite the config for all of them.